sormas-backend/src/main/java/de/symeda/sormas/backend/common/AbstractCoreAdoService.java

```diff
@@ -1,31 +1,40 @@
+/*
+ * SORMAS® - Surveillance Outbreak Response Management & Analysis System
+ * Copyright © 2016-2021 Helmholtz-Zentrum für Infektionsforschung GmbH (HZI)
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
 package de.symeda.sormas.backend.common;
 
-import java.sql.Timestamp;
+import de.symeda.sormas.backend.travelentry.TravelEntry;
 
 import javax.persistence.criteria.CriteriaBuilder;
-import javax.persistence.criteria.From;
-import javax.persistence.criteria.JoinType;
-import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
 
-public abstract class AbstractCoreAdoService<ADO extends CoreAdo> extends AdoServiceWithUserFilter<ADO> {
+public abstract class AbstractCoreAdoService<ADO extends CoreAdo> extends AbstractDeletableAdoService<ADO> {
 
 	public AbstractCoreAdoService(Class<ADO> elementClass) {
 		super(elementClass);
 	}
 
-	@Override
-	public void delete(ADO deleteme) {
-
-		deleteme.setDeleted(true);
-		em.persist(deleteme);
-		em.flush();
-	}
+	public boolean isArchived(String uuid) {
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Long> cq = cb.createQuery(Long.class);
+		Root<ADO> from = cq.from(getElementClass());
 
-	protected <C> Predicate changeDateFilter(CriteriaBuilder cb, Timestamp date, From<?, C> path, String... joinFields) {
-		From<?, ?> parent = path;
-		for (int i = 0; i < joinFields.length; i++) {
-			parent = parent.join(joinFields[i], JoinType.LEFT);
-		}
-		return CriteriaBuilderHelper.greaterThanAndNotNull(cb, parent.get(AbstractDomainObject.CHANGE_DATE), date);
+		cq.where(cb.and(cb.equal(from.get(CoreAdo.ARCHIVED), true), cb.equal(from.get(AbstractDomainObject.UUID), uuid)));
+		cq.select(cb.count(from));
+		long count = em.createQuery(cq).getSingleResult();
+		return count > 0;
 	}
```

Comment:
This class should contain both `isArchived` and `isDeleted()` method similar to the one in `TravelEntryService`. Once this is done, check for useless overwrites. Probably this service method can be generically forwarded to the ejb.

Reply:
deleted part will come with different ticket
i know it makes sense to do it but not now
evPart for example has no deleted flag right now